src/components/Tree/TreeItem.tsx

```diff
@@ -0,0 +1,147 @@
+
+    const [expanded, setExpanded] = useState<boolean>(false);
+    const [hovered, setHovered] = useState<boolean>(false);
+
+    const handleMouseEnter = () => setHovered(true);
+    const handleMouseLeave = () => setHovered(false);
+
+    const [{ opacity }, drag] = useDrag({
+        item: { id, sort },
+        collect: (monitor) => ({
+            opacity: monitor.isDragging() ? 0.4 : 1,
+        }),
+        type: treeId,
+        canDrag: draggable,
+    });
+
+    const handleDrop = (
+        targetItem: DraggableItem<{ id: string; sort: Nullable<number> }>,
+        sourceItem: DraggableItem<{ id: string; sort: Nullable<number> }>,
+        position: DropZonePosition,
+    ) => {
+        onDrop?.(targetItem, sourceItem, position);
+    };
+
+    const handleSelect = () => {
+        onItemSelect(id);
+        onSelect?.(id);
+    };
+
+    const toggleExpand = (event: React.MouseEvent<HTMLDivElement, MouseEvent>) => {
+        event.stopPropagation();
+        setExpanded((previousExpanded) => !previousExpanded);
+    };
+
+    const childrenArray = React.Children.toArray(children);
+
+    let enhancedChildren: ReactNode = childrenArray;
+
+    if (draggable) {
+        enhancedChildren = React.Children.map(children, (child, index) => {
+            if (!child) {
+                return <></>;
+            }
+
+            return React.cloneElement(
+                <>
+                    {index === 0 && (
+                        <DropZone
+                            data={{
+                                targetItem: { id, sort },
+                                position: DropZonePosition.Before,
+                            }}
+                            onDrop={handleDrop}
+                            treeId={treeId}
+                        />
+                    )}
+
+                    {child}
+                </>,
+            );
+        });
+    }
+
+    const caretComponent =
+        childrenArray.length > 0 ? (
+            <div
+                className={merge([
+                    'tw-transition-transform tw-w-0 tw-h-0 tw-text-black-100 tw-text-opacity-40 tw-font-normal tw-border-t-4 tw-border-t-transparent tw-border-b-4 tw-border-b-transparent tw-border-l-4 tw-border-l-x-strong',
+                    expanded ? 'tw-rotate-90' : '',
+                    selectedIds.includes(id) && 'tw-text-box-selected-strong-inverse',
+                ])}
+            />
+        ) : (
+            <></>
+        );
+
+    return (
+        <li data-test-id="tree-item" ref={drag} style={{ opacity }}>
+            <DropZone
+                data={{
+                    targetItem: { id, sort },
+                    position: DropZonePosition.Within,
+                }}
+                treeId={treeId}
+            >
+                {/* eslint-disable-next-line jsx-a11y/click-events-have-key-events, jsx-a11y/no-static-element-interactions */}
```

Comment:
We shouldn't forget accessibility, is it a false positive?

Reply:
Accessibility is planned for the next phase. The old one also wasn't very accessible so nothing lost nothing gained, but it's definitely something we'll do in the near future.